src/App.tsx

```diff
@@ -1,122 +1,27 @@
 import './App.css';
 import './css_reset.css'
 import './section_view.css';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faFaceSmile } from '@fortawesome/free-solid-svg-icons';
+import * as Types from './types';
+import { Files } from './Files';
+import { ChordProgression } from './ChordProgression';
+import { Comments } from './Comments';
+import { CreateComment } from './CreateComment';
+import { SectionTitle } from './SectionTitle';
+import { useState } from 'react';
```

Comment:
I'd like to add some linting to allow us to have a consistent ordering of imports. Then you can configure vscode to perform any lint fixes every time the file is saved locally

Reply:
Okay, so what about integrating ESLint or Prettier in my repo? That's one option, but I'm curious what approach you think is best

Reply:
Yeah you can do it yourself as well. I would focus on eslint out of those two. I've been trying out biomejs but it's a bit too opinionated to me and doesn't allow for enough customization. Eslint supports all rules you could ever want, plus any styling that is supported by prettier for the most part. Eslint is king

Reply:
How about we talk about this over video? I feel like there's a fair bit to unpack in this series of comments and I want to be sure we're on the same page

Reply:
Here's the things I'm trying to figure out about this:




1. What’s Promise<SectionPageAllData> ?  

How do I know what the shape of this type will need to look like when I’m writing something similar in the future?

I guess the type here is for an object which contains all the data from localstorage/backend?





2. How are these two functions related? One seems to be for adding a comment and the other is for getting a file from localstorage. I get that they’re both useful but it feels kind of arbitrary to store them both together since they don’t seem related other than both making use of our mock backend. Maybe that’s the reason





3. The first function almost feels redundant. The function call it’s returning has essentially the same name as it. So what’s the point of it?


4. I’m not sure what you mean that client refers to http client. I know what a client is, but I’m not understanding the significance of this statement

5. I'd like to know more about what you have in mind for the project page with different views

Reply:
I'm thinking about this, and I think that one large appeal of the code you wrote is that we're already giving things types such they could be async, despite localStorage being synchronous, so that when we go go to an actual backend we don't have to change them.

Would love to know your thoughts!

Reply:
> I think that one large appeal of the code you wrote is that we're already giving things types such they could be async, despite localStorage being synchronous, so that when we go go to an actual backend we don't have to change them.

@aelishRollo Yes that's correct :+1: You got it perfectly. We should "act" like we are working with a real backend as much as possible, including the naming and usage of functions being agnostic to which "backend" we are using, so we can then swap them out from each other. It may make even more sense to create an `interface`, and have a `class LocalStorageBackendAPI` that implements that interface.

As far as the term "backend", you could call the `localStorage` in this case a "storage backend", so it kind of already is a backend in itself.